yabridge: misbehaving plugins

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

Plugins that used to work just fine are now misbehaving. 

Rolling Sampler: Drag and Drop no longer works.(EDIT: It seems to work I just had it pointed to a folder with bad permissions)
MIDI CAP: No longer allow me to interact with the window.
Minimal Audio Plugins: The GUI does not align with mouse clicks(possibly why MIDI CAP doesn't work).  Clicking on the top left of the GUI activates a button on a completely different part of the GUI.
Reason Rack: The plugin window is now black even though DXVK is installed(it used to work just fine).


### Steps to reproduce

These plugins used to have no issues! Using multiple versions of Bitwig and WINE makes no difference. 

### Expected behaviour

Functionality above to work.

### Screenshots

_No response_

### Relevant log output

```console
I'm not sure what logs I may need or how to gather the relevant ones.
```

### Additional context

I'm using Wayland.
I haven't used a different version of Yabridge.
I used Bitwigs newest beta 5.3b5 and the latest public version 5.2.7.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.65-rt47, NixOS, 25.05 (Warbler), 25.05.git.bb360ca74M`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@kira-bruneau
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Here's my workaround, I have no idea why this works to be honest:

- Copy yabridge and yabridgectl directories from nixpkgs somewhere locally
- Replace `wine` with `wineWowPackages.staging` in yabridge/default.nix AND yabridgectl/default.nix
- sudo nixos-rebuild switch --flake .#myCoolFlake
- Sync VSTs
- Profit ?

yabridge/default.nix:
```diff
diff --git a/pkgs/tools/audio/yabridge/default.nix b/pkgs/tools/audio/yabridge/default.nix
index ef97b99163f3..7f058e0c7b7f 100644
--- a/pkgs/tools/audio/yabridge/default.nix
+++ b/pkgs/tools/audio/yabridge/default.nix
@@ -8,7 +8,7 @@
   meson,
   ninja,
   pkg-config,
-  wine,
+  wineWowPackages,
   libxcb,
   nix-update-script,
 }:
@@ -102,7 +102,7 @@ multiStdenv.mkDerivation (finalAttrs: {
     (replaceVars ./hardcode-dependencies.patch {
       libdbus = dbus.lib;
       libxcb32 = pkgsi686Linux.xorg.libxcb;
-      inherit wine;
+      wine = wineWowPackages.staging;
     })
 
     # Patch the chainloader to search for libyabridge through NIX_PROFILES
@@ -125,7 +125,7 @@ multiStdenv.mkDerivation (finalAttrs: {
     meson
     ninja
     pkg-config
-    wine
+    wineWowPackages.staging
   ];
 
   buildInputs = [
@@ -153,10 +153,10 @@ multiStdenv.mkDerivation (finalAttrs: {
   # Hard code wine path in wrapper scripts generated by winegcc
   postFixup = ''
     substituteInPlace "$out/bin/yabridge-host-32.exe" \
-      --replace 'WINELOADER="wine"' 'WINELOADER="${wine}/bin/wine"'
+      --replace 'WINELOADER="wine"' 'WINELOADER="${wineWowPackages.staging}/bin/wine"'
 
     substituteInPlace "$out/bin/yabridge-host.exe" \
-      --replace 'WINELOADER="wine"' 'WINELOADER="${wine}/bin/wine64"'
+      --replace 'WINELOADER="wine"' 'WINELOADER="${wineWowPackages.staging}/bin/wine64"'
   '';
 
   passthru.updateScript = nix-update-script { };
```

yabridgectl/default.nix:
```diff
diff --git a/pkgs/tools/audio/yabridgectl/default.nix b/pkgs/tools/audio/yabridgectl/default.nix
index 4e69dc8462da..983f8bbe0d73 100644
--- a/pkgs/tools/audio/yabridgectl/default.nix
+++ b/pkgs/tools/audio/yabridgectl/default.nix
@@ -1,11 +1,15 @@
 {
   lib,
   rustPlatform,
-  yabridge,
   makeWrapper,
-  wine,
+  wineWowPackages,
+  callPackage,
 }:
 
+let
+  yabridge = callPackage ../yabridge {
+  };
+in
 rustPlatform.buildRustPackage {
   pname = "yabridgectl";
   version = yabridge.version;
@@ -35,7 +39,7 @@ rustPlatform.buildRustPackage {
     wrapProgram "$out/bin/yabridgectl" \
       --prefix PATH : ${
         lib.makeBinPath [
-          wine # winedump
+          wineWowPackages.staging # winedump
         ]
       }
   '';
```

Then in your flake:

  yabridge = pkgs.callPackage ./custom/yabridge { };
  yabridgectl = pkgs.callPackage ./custom/yabridgectl { };

...

    home.packages = [
      yabridge
      yabridgectl
];

Sync VSTs:
`WINEPREFIX="/tmp/testpfx" yabridgectl sync --prune`

Also, it's possible to use custom wine builds to get a wine version under which yabridge was properly tested by running yabridgectl/your DAW with steam-run:
`WINEPREFIX="/tmp/testpfx" WINELOADER=/tmp/wine-9.21-amd64/bin/wine WINESERVER=/tmp/wine-9.21-amd64/bin/wineserver steam-run yabridgectl sync --prune`

If wine is not replaced with wineWowPackages.staging, compilation fails:
`[428/436] Linking target yabridge-host
FAILED: yabridge-host 
wineg++  -o yabridge-host yabridge-host.p/src_common_communication_vst2.cpp.o yabridge-host.p/src_common_serialization_vst2.cpp.o yabridge-host.p/src_common_configuration.cpp.o yabridge-host.p/src_common_logging_common.cpp.o yabridge-host.p/src_common_logging_vst2.cp>
/nix/store/vk4mlknqk9yjbqa68a7rvpfxfdw3rad7-binutils-2.43.1/bin/ld: cannot find -lole32: No such file or directory
/nix/store/vk4mlknqk9yjbqa68a7rvpfxfdw3rad7-binutils-2.43.1/bin/ld: cannot find -lshell32: No such file or directory
/nix/store/vk4mlknqk9yjbqa68a7rvpfxfdw3rad7-binutils-2.43.1/bin/ld: cannot find -luuid: No such file or directory
collect2: error: ld returned 1 exit status`

--- a/pkgs/applications/emulators/wine/sources.nix
+++ b/pkgs/applications/emulators/wine/sources.nix
@@ -24,9 +24,9 @@ let fetchurl = args@{url, hash, ...}:
 in rec {
 
   stable = fetchurl rec {
-    version = "9.0";
-    url = "https://dl.winehq.org/wine/source/9.0/wine-${version}.tar.xz";
-    hash = "sha256-fP0JClOV9bdtlbtd76yKMSyN5MBwwRY7i1jaODMMpu4=";
+    version = "10.0";
+    url = "https://dl.winehq.org/wine/source/10.0/wine-${version}.tar.xz";
+    hash = "sha256-xeCz9ffvr7MOnNTZxiS4XFgxcdM1SdkzzTQC80GsNgE=";
 
     ## see http://wiki.winehq.org/Gecko
     gecko32 = fetchurl rec {
@@ -69,9 +69,9 @@ in rec {
 
   unstable = fetchurl rec {
     # NOTE: Don't forget to change the hash for staging as well.
-    version = "10.0-rc5";
+    version = "10.0";
     url = "https://dl.winehq.org/wine/source/10.0/wine-${version}.tar.xz";
-    hash = "sha256-7IlPe1wH0tTCHVNNf0OUZnuO8ngH4w4hKJhpp02XbtI=";
+    hash = "sha256-xeCz9ffvr7MOnNTZxiS4XFgxcdM1SdkzzTQC80GsNgE=";
     inherit (stable) patches;
 
     ## see http://wiki.winehq.org/Gecko
@@ -117,7 +117,7 @@ in rec {
   staging = fetchFromGitLab rec {
     # https://gitlab.winehq.org/wine/wine-staging
     inherit (unstable) version;
-    hash = "sha256-no+yYF/xhy0kRfBqPer0UbpJNEh9LtKCmyVvhQB58K4=";
+    hash = "sha256-0mzKoaNaJ6ZDYQtJFU383W5nNe/FKtpBjeWDpiqkmp4=";
     domain = "gitlab.winehq.org";
     owner = "wine";
     repo = "wine-staging";
@@ -126,35 +126,7 @@ in rec {
     disabledPatchsets = [ ];
   };
 
-  wayland = fetchFromGitLab {
-    # https://gitlab.collabora.com/alf/wine/-/tree/wayland
-    version = "8.2";
-    hash = "sha256-Eb2SFBIeQQ3cVZkUQcwNT5mcYe0ShFxBdMc3BlqkwTo=";
-    domain = "gitlab.collabora.com";
-    owner = "alf";
-    repo = "wine";
-    rev = "b2547ddf9e08cafce98cf7734d5c4ec926ef3536";
-
-    inherit (unstable) gecko32 gecko64;
-
-    inherit (unstable) mono;
-
-    updateScript = writeShellScript "update-wine-wayland" ''
-      ${updateScriptPreamble}
-      wayland_rev=$(get_source_attr wayland.rev)
-
-      latest_wayland_rev=$(curl -s 'https://gitlab.collabora.com/api/v4/projects/2847/repository/branches/wayland' | jq -r .commit.id)
-
-      if [[ "$wayland_rev" != "$latest_wayland_rev" ]]; then
-          latest_wayland=$(curl -s 'https://gitlab.collabora.com/alf/wine/-/raw/wayland/VERSION' | cut -f3 -d' ')
-          wayland_url=$(get_source_attr wayland.url)
-          set_version_and_hash wayland "$latest_wayland" "$(nix-prefetch-url --unpack "''${wayland_url/$wayland_rev/$latest_wayland_rev}")"
-          set_source_attr wayland rev "\"$latest_wayland_rev\""
-      fi
-
-      do_update
-    '';
-  };
+  wayland = pkgs.lib.warnOnInstantiate "building wine with `wineRelease = \"wayland\"` is deprecated. Wine now builds with the wayland driver by default." stable; # added 2025-01-23
 
   winetricks = fetchFromGitHub rec {
     # https://github.com/Winetricks/winetricks/releases

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -18107,7 +18107,7 @@ with pkgs;
   });
 
   wine-wayland = lowPrio (winePackages.full.override {
-    wineRelease = "wayland";
+    x11Support = false;
   });
 
   inherit (callPackage ../servers/web-apps/wordpress {})

--- a/pkgs/top-level/wine-packages.nix
+++ b/pkgs/top-level/wine-packages.nix
@@ -58,11 +58,9 @@ rec {
   stagingFull = full.override { wineRelease = "staging"; };
 
   wayland = base.override {
-    wineRelease = "wayland";
-    waylandSupport = true;
+    x11Support = false;
   };
   waylandFull = full.override {
-    wineRelease = "wayland";
-    waylandSupport = true;
+    x11Support = false;
   };
 }

